src/main/query_profiler.cpp

```diff
@@ -263,24 +268,36 @@ void QueryProfiler::EndPhase() {
 	}
 }
 
-OperatorProfiler::OperatorProfiler(ClientContext &context) {
-	enabled = QueryProfiler::Get(context).IsEnabled();
-	settings = ClientConfig::GetConfig(context).profiler_settings;
-}
-
-bool OperatorProfiler::SettingIsEnabled(MetricsType metric) const {
+bool SettingIsEnabled(profiler_settings_t settings, MetricsType metric) {
 	if (settings.find(metric) != settings.end()) {
 		return true;
 	}
-	if (metric == MetricsType::OPERATOR_TIMING && SettingIsEnabled(MetricsType::CPU_TIME)) {
-		return true;
-	}
-	if (metric == MetricsType::OPERATOR_CARDINALITY && SettingIsEnabled(MetricsType::CUMULATIVE_CARDINALITY)) {
-		return true;
+
+	switch (metric) {
+	case MetricsType::OPERATOR_TIMING:
+		return SettingIsEnabled(settings, MetricsType::CPU_TIME);
+	case MetricsType::OPERATOR_CARDINALITY:
+		return SettingIsEnabled(settings, MetricsType::CUMULATIVE_CARDINALITY);
+	case MetricsType::OPERATOR_ROWS_SCANNED:
+		return SettingIsEnabled(settings, MetricsType::CUMULATIVE_ROWS_SCANNED);
+	default:
+		break;
 	}
+
 	return false;
 }
 
+OperatorProfiler::OperatorProfiler(ClientContext &context) : context(context) {
+	enabled = QueryProfiler::Get(context).IsEnabled();
+	auto settings = ClientConfig::GetConfig(context).profiler_settings;
+
+	vector<MetricsType> op_metrics = {MetricsType::OPERATOR_TIMING, MetricsType::OPERATOR_CARDINALITY,
+	                                  MetricsType::OPERATOR_ROWS_SCANNED};
```

Comment:
I'm unsure how robust this is. I.e., how easy it is to forget to add a setting here. Maybe we can add a static function to `ProfilingInfo` similar to the default settings function?
```cpp
profiler_settings_t ProfilingInfo::DefaultSettings() {
	return {
	    MetricsType::CPU_TIME,
	    MetricsType::EXTRA_INFO,
	    MetricsType::CUMULATIVE_CARDINALITY,
	    MetricsType::OPERATOR_CARDINALITY,
	    MetricsType::OPERATOR_TIMING,
	};
}
```